### PR TITLE
Add server pfx certificate and quic client token validation

### PR DESCRIPTION
**Pull request by qwu16**

### CODE REVIEW

File: source/agent/conference/conference.js [hunk 1]
```diff
@@ -108,6 +108,8 @@ var Conference = function (rpcClient, selfRpcId) {
   var rtcController;
   let quicController;
 
+  var roomToken = Math.round(Math.random() * 1000000000000000000) + '';
```
Comment (starwarfan) [inline source/agent/conference/conference.js:111]: Change random function

Comment (qwu16) [reply source/agent/conference/conference.js]: use crypto.randomBytes to generate secure random bytes

File: source/cascading_control/cloud_service.js [hunk 2]
```diff
@@ -0,0 +1,426 @@
+// MIT License
+//
+// Copyright (c) 2012 Universidad Politécnica de Madrid
```
Comment (starwarfan) [inline source/cascading_control/cloud_service.js:3]: license

File: source/cascading_control/cloud_service.js [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    return last_used[region];
+  } else {
+    const keys = Object.keys(clusters);
+    const random = Math.floor(Math.random() * keys.length);
+    
+    console.log(random, keys[random]);
+    last_used[region] = keys[random];
+    console.log("selected cluster:", keys[random]);
+    return keys[random];
+    }
```
Comment (starwarfan) [inline source/cascading_control/cloud_service.js:130]: strange indent

File: source/cascading_control/cloud_service.js [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+  var conferenceID = Math.round(Math.random() * 100000000000000000000) + '';
+  res.send(conferenceID);
+});
+
+/*
+ *Maybe there are multiple clusters in the same region, here we select last used cluster in the same region
+ *different shedule polices can be customized for different scenarios
+ */
+var selectCluster = function (region, clusters) {
+  console.log("select cluster for region:", region, " last used:", last_used);
```
Comment (starwarfan) [inline source/cascading_control/cloud_service.js:119]: inconsistent string quotation mark

File: source/cascading_control/restClient.js [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+      return;
+    }
+    send('GET', 'rooms/' + room, undefined, function(roomRtn) {
+      var room = JSON.parse(roomRtn);
+      callback(room);
+    }, callbackError);
+  };
+
+  that.createToken = function(room, user, role, preference, callback, callbackError) {
+    /*console.log("typeof room:", typeof room);
```
Comment (starwarfan) [inline source/cascading_control/restClient.js:322]: clean up

File: source/cluster_manager/clusterManager.js [hunk 6]
```diff
@@ -60,8 +60,9 @@ var ClusterManager = function (clusterName, selfId, spec) {
           'Content-Length': data.length,
         },
       };
+      ssl && (options.rejectUnauthorized = false);
       log.info("send options:", options);
-
+      const http = (ssl ? require('https'): require('http'));
```
Comment (starwarfan) [inline source/cluster_manager/clusterManager.js:65]: inconsistent spaces for `? : `

File: source/cascading_control/cloud_service.js [hunk 7]
```diff
@@ -0,0 +1,426 @@
+// MIT License
```
Comment (starwarfan) [inline source/cascading_control/cloud_service.js:1]: I recommend to rename it to some sample. Since this file has some s issues.

**Review by starwarfan — APPROVED** [review]

**Pull request merged**